src/Symfony/Component/HttpKernel/Controller/TraceableControllerResolver.php

```diff
@@ -52,12 +59,20 @@ public function getController(Request $request)
 
     /**
      * {@inheritdoc}
+     *
+     * @deprecated This method is deprecated as of 3.1 and will be removed in 4.0.
      */
     public function getArguments(Request $request, $controller)
     {
+        @trigger_error(sprintf('The %s method is deprecated as of 3.1 and will be removed in 4.0. Please use the %s instead.', __METHOD__, TraceableArgumentResolver::class), E_USER_DEPRECATED);
+
+        if ($this->argumentResolver instanceof TraceableArgumentResolver) {
```

Comment:
even better could be to always wrap the resolver in a traceable one if not done already, to avoid duplicating the logic.